### PR TITLE
Screaming Snake Case naming convention

### PR DESCRIPTION
**Pull request by tabdulradi**
Stills needs test cases, but wanted to make sure first if it's a welcomed addition?

### CODE REVIEW

**Comment by jcazevedo** [comment]
I welcome this addition if it is used in the wild, as it seems to be by looking at [Wikipedia](https://en.wikipedia.org/wiki/Snake_case) and [Stack Overflow](https://stackoverflow.com/questions/11273282/whats-the-name-for-hyphen-separated-case)! I confess I didn't know about the "Screaming Snake Case" term nor have found uses of it in Scala projects yet.

Thanks @tabdulradi!

**Comment by leifwickland** [comment]
@jcazevedo I'm not sure that I've heard anyone use the term within the Scala community, but I am familiar with the term from other contexts.

I'm disappointed to report that I have needed this convention in the past to match the style from a project in another language.

I favor adding this feature.

**Comment by leifwickland** [comment]
@tabdulradi,

Thanks for the contribution.

Please add the new convention to the [list of conventions in the documentation](https://github.com/pureconfig/pureconfig/blame/master/docs/docs/docs/overriding-behavior-for-case-classes.md#L64)

**Comment by ruippeixotog** [comment]
I haven't heard that term before either, but I'm not surprised there's a name for it and I'm actually surprised that this feature request didn't come up earlier, given that Typesafe Config injects env vars in configs 🙂

File: docs/docs/docs/overriding-behavior-for-case-classes.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ library:
 
 * [`CamelCase`](https://en.wikipedia.org/wiki/Camel_case) (examples: `camelCase`, `useMorePureconfig`);
 * [`SnakeCase`](https://en.wikipedia.org/wiki/Snake_case) (examples: `snake_case`, `use_more_pureconfig`);
+* ScreamingSnakeCase (examples: `SCREAMING_SNAKE_CASE`, `USE_MORE_PURECONFIG`);
```
Comment (ruippeixotog) [inline docs/docs/docs/overriding-behavior-for-case-classes.md]: nit: can you make this monospace and link to https://en.wikipedia.org/wiki/Snake_case? I think it's still useful to link to it even if it's only mentioned in the History section.

```suggestion
* [`ScreamingSnakeCase`](https://en.wikipedia.org/wiki/Snake_case) (examples: `SCREAMING_SNAKE_CASE`, `USE_MORE_PURECONFIG`);
```

**Review by ruippeixotog — CHANGES_REQUESTED** [review]
LGTM! Just a nit below.

**Review by ruippeixotog — APPROVED** [review]

**Comment by ruippeixotog** [comment]
Thanks!

**Pull request merged**